addon/doxysearch/CMakeLists.txt

```diff
@@ -1,12 +1,12 @@
-find_package(Xapian REQUIRED)
+find_package(xapian REQUIRED)
```

Comment:
You changed `Xapian` to `xapian` for the package with the reason:
> So the casing of the cmake package is the same as the one from upstream xapian-core.

but in principle a package name doesn't have to have the same spelling as the directories used.
Internally in `FindXapian.cmake` we see that for directories etc `xapian` is used.

Furthermore on the Xapian site (https://xapian.org/) we see that in sentences also `Xapian` is used, like:
> Welcome to the Xapian project website.

Secondly this change has nothing to do with the "Allow building doxygen as a subproject" so mixing is is not really a good idea.

Reply:
I'll remove the xapian related change and open a new issue about the xapian thing.

In the mean time, see https://oligarchy.co.uk/xapian/1.4.17/xapian-core-1.4.17.tar.xz.
It contains `cmake/xapian-config.cmake.in`.

Reply:
The mentioned file has indeed lowercase `xapian` in its name but inside I see e.g.
```
SET(XAPIAN_LIBRARIES "@libdir@/libxapian@LIBRARY_VERSION_SUFFIX@@SHLIBEXT@" CACHE FILEPATH "Libraries for Xapian")
SET(XAPIAN_INCLUDE_DIR "@incdir@" CACHE PATH "Include path for Xapian")
```

Reply:
I've opened https://github.com/doxygen/doxygen/issues/8038

It's the casing of the file that matters.